### PR TITLE
Adding AppInfo Struct to RegisterAppInterface

### LINKED ISSUE #398: Update SDK to include AppInfo struct parameter in RAI request
We need to update the SDK to include the new AppInfo struct that includes a bundleID used for remote app launching strategies.  Please see attached xml doc for details.
[AppLink_Mobile_API_v4_x _withAppInfo.zip](https://github.com/smartdevicelink/sdl_ios/files/263271/AppLink_Mobile_API_v4_x._withAppInfo.zip)

### PR DESCRIPTION
**Pull request by asm09fsu**
Fixes #398 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Tests have been added to SDLRegisterAppInterfaceSpec and SDLRPCRequestFactorySpec and newly made SDLAppInfoSpec.

### Summary
This PR adds the ability to send an app info struct with a register app interface to assist in remote app launching strategies.

##### Enchancements
* Added ability to send app info with a register app interface to assist in remote app launching strategies.

### CODE REVIEW

**Comment by asm09fsu** [comment]
@joeljfischer please review

File: SmartDeviceLink/SDLRPCRequestFactory.m [hunk 1]
```diff
@@ -393,6 +398,15 @@ + (SDLDeviceInfo *)sdl_buildDeviceInfo {
     return deviceInfo;
 }
 
++ (SDLAppInfo*)sdl_buildAppInfo {
+    SDLAppInfo* appInfo = [[SDLAppInfo alloc] init];
+    NSBundle* mainBundle = [NSBundle mainBundle];
+    NSDictionary* bundleDictionary = mainBundle.infoDictionary;
+    appInfo.appVersion = [bundleDictionary objectForKey:kBundleShortVersionStringKey];
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLRPCRequestFactory.m]: This can use shortened syntax: `bundleDictionary[kBundleShortVersionStringKey]`.

**Comment by joeljfischer** [comment]
@justinjdickow where is this on implementation into Core?

**Comment by joeljfischer** [comment]
LGTM

**Pull request merged**